scipy/signal/spectral.py

```diff
@@ -697,36 +1262,35 @@ def _spectral_helper(x, y, fs=1.0, window='hann', nperseg=256,
     References
     ----------
     .. [1] Stack Overflow, "Rolling window for 1D arrays in Numpy?",
-        http://stackoverflow.com/a/6811241
-    .. [2] Stack Overflow, "Using strides for an efficient moving average
-        filter", http://stackoverflow.com/a/4947453
+           http://stackoverflow.com/a/6811241
+    .. [2] Stack Overflow, "Using strides for an efficient moving
+           average filter", http://stackoverflow.com/a/4947453
 
     Notes
     -----
     Adapted from matplotlib.mlab
 
     .. versionadded:: 0.16.0
     """
-    if mode not in ['psd', 'complex', 'magnitude', 'angle', 'phase']:
```

Comment:
Okay found it: https://github.com/scipy/scipy/issues/5757#issuecomment-191617002. The comments above talk about keeping ``mode-'complex'``, but the linked comment proposes to replace that with ``stft``.

The comment above it from @endolith about derivative of phase and ampl/phase being required for some applications is still not addressed then though. If those modes are reimplemented as you suggest by deriving them correctly from the output of ``stft``, would that be done within ``spectrogram``? If so no deprecation is needed, but the modes should not be removed and the implementation simply fixed (returning different values, which is OK given that it's a bug fix).